src/hdf5_interface.F90

```diff
@@ -2347,6 +2426,67 @@ subroutine read_attribute_string(buffer, obj_id, name)
     call h5tclose_f(memtype, hdf5_err)
   end subroutine read_attribute_string
 
+  subroutine read_attribute_string_1D(buffer, obj_id, name)
+    character(*), target, allocatable, intent(inout) :: buffer(:)
+    integer(HID_T), intent(in) :: obj_id
+    character(*),   intent(in) :: name
+
+    integer          :: hdf5_err
+    integer(HID_T)   :: space_id
+    integer(HID_T)   :: attr_id
+    integer(HSIZE_T) :: dims(1)
+    integer(HSIZE_T) :: maxdims(1)
+
+    call h5aopen_f(obj_id, trim(name), attr_id, hdf5_err)
+
+    if (allocated(buffer)) then
+      dims(:) = shape(buffer)
+    else
+      call h5aget_space_f(attr_id, space_id, hdf5_err)
+      call h5sget_simple_extent_dims_f(space_id, dims, maxdims, hdf5_err)
```

Comment:
GCC 5.3.0 and parallel HDF5-1.8.16

The model I've been testing with is test_iso_in_lab.py.  The trouble comes from this guy:
```
HDF5 "c_H_in_H2O.h5" {
GROUP "/" {
   GROUP "c_H_in_H2O" {
      ATTRIBUTE "nuclides" {
         DATATYPE  H5T_STRING {
            STRSIZE 6;
            STRPAD H5T_STR_NULLPAD;
            CSET H5T_CSET_ASCII;
            CTYPE H5T_C_S1;
         }
         DATASPACE  SCALAR
         DATA {
         (0): "['H1']"
         }
      }
```